tests/ctst/world/Zenko.ts

```diff
@@ -99,6 +106,318 @@ export default class Zenko extends World {
+                            role.Arn.includes(ARWWITargetRole) && role.Arn.includes(account.id),
+                    )?.Arn || roleToAssume;
+                });
+            }
+            // Ensure we can assume at least one role
+            if (!roleToAssume) {
+                throw new Error('Error when trying to list roles for web identity.');
+            }
+            // Arn to assume
+            const arn = roleToAssume;
+            this.options.roleArn = arn;
+            // Assume the role and save the credentials
+            const ARWWI = await STS.assumeRoleWithWebIdentity(this.options, this.parameters);
+            if (ARWWI && typeof ARWWI !== 'string' && ARWWI.stdout) {
+                this.parameters.AssumedSession = JSON.parse(ARWWI.stdout).Credentials;
+            } else {
+                throw new Error('Error when trying to Assume Role With Web Identity.');
+            }
+            // Save the session for future scenarios (increases performance)
+            CacheHelper.ARWWI[ARWWIName] = this.parameters.AssumedSession;
+            this.cliMode.parameters.AssumedSession = CacheHelper.ARWWI[ARWWIName];
+            this.cliMode.assumed = true;
+        } else {
+            this.parameters.AssumedSession = CacheHelper.ARWWI[ARWWIName];
+            this.cliMode.parameters.AssumedSession = CacheHelper.ARWWI[ARWWIName];
+            this.cliMode.assumed = true;
+        }
+    }
+
+    /**
+     * HTTP client to request JWT token given the username and password.
+     *
+     * @param {string} username - username of user requesting token
+     * @param {string} password - password of user requesting token
+     * @param {string} host - host URL of keycloak service
+     * @param {number} port - port of keycloak service
+     * @param {string} path - path of keycloak service authentication API
+     * @param {string} clientId - id of the client of the user
+     * @param {string} grantType - grant of the user
+     * @returns {string} the OIDC token
+     */
+    async getWebIdentityToken(
+        username: string,
+        password: string,
+        host: string,
+        port: string,
+        path: string,
+        clientId: string,
+        grantType: string,
+    ) {
+        const baseUrl = this.parameters.ssl === false ? 'http://' : 'https://';
+        const data = qs.stringify({
+            username,
+            password,
+            // eslint-disable-next-line camelcase
+            client_id: clientId,
+            // eslint-disable-next-line camelcase
+            grant_type: grantType,
+        });
+        const config: AxiosRequestConfig = {
+            method: 'post',
+            url: `${baseUrl}${host}:${port}${path}`,
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            data,
+        };
+        const result: AxiosResponse = await axios(config);
+        return result.data.access_token;
```

Comment:
Usually the approach for tests, from my experience, is "let it throw", as it will return an error anyway...

It may be better to throw a known and human-readable error though, to ease the debugging, as long as enough information is logged along with the message

Reply:
yeah! aiming for easy debugging as well.
If you think error is clear enough, we should be good.
Feel free to resolve the comment if that's the case.

Reply:
perfect!